packages/npm-dependencies-validation/src/utils/npmUtil.ts

```diff
@@ -1,23 +1,46 @@
 import { spawn } from "child_process";
+import { NpmCommandConfig, VscodeOutputChannel } from "../types";
 
 export function getNPM(): string {
   return /^win/.test(process.platform) ? "npm.cmd" : "npm";
 }
 
 export function invokeNPMCommand<T>(
-  commandArgs: string[],
-  cwd: string
+  config: NpmCommandConfig,
+  outputChannel?: VscodeOutputChannel
 ): Promise<T> {
+  const { commandArgs, cwd } = config;
   return new Promise((resolve, reject) => {
     const command = spawn(getNPM(), [...commandArgs, "--json"], { cwd });
 
     command.stdout.on("data", (data) => {
+      showData(`${data}`, outputChannel);
```

Comment:
is there a max buffer size here when we spawn the process? 
will we always get the data from JSON.parse in one big batch?

Reply:
it is always one big chunk.
I have tested it on relatively big project and it worked